File: misc/convert-binary-string-to-decimal/node-18.12/solution1.js
module.exports = function toDecimal(binary) {
  return binary
    .split(``)
    .map((d, i, arr) => (d === `0` ? 0 : 2 ** (arr.length - 1 - i)))
    .reduce((total, e) => total + e, 0);
};

File: misc/convert-binary-string-to-decimal/dart-2.18/solution1.dart
import 'dart:math';

int toDecimal(String binary) {
  final List<String> digits = binary.split('');

  return digits
      .asMap()
      .entries
      .map(
        (MapEntry<int, String> e) =>
            e.value == '0' ? 0 : pow(2, digits.length - 1 - e.key) ~/ 1,
      )
      .reduce((int value, int element) => value + element);
}
